lib/storage/errors.rb

```diff
@@ -28,17 +28,38 @@ module Storage
   StorageError = Class.new(RuntimeError)
   AbstractMethodError = Class.new(StandardError)
 
+  class InvalidCredentialsError < StandardError
+    def initialize(provider)
+      msg = "Invalid credentials given for provider '#{provider}'"
```

Comment:
Might be a nice idea in future to have access to the `friendly_name` provider attribute to use for this error message?

Reply:
If we can find a nice way to do it, maybe. Changing where `ClientFactory::PROVIDERS` is defined? Although I don't think an error class should need to access anything other than the string that it's been explicitly given.

Reply:
Maybe `friendly_name` should be an attribute of `client` so that providers can access it? With it being part of `ClientFactory` i don't see it being used much, and I agree that the error class itself shouldn't do anything more than print what it's given, but if we're going to have `friendly_name` in the first place then this may as well  be where we use it.

Reply:
I've just pushed a commit that moves the friendly name definition to the provider class. It's still a part of the `ClientFactory::PROVIDERS` hash, but its value is fetched from the provider class.